File: C++ Kasus/src/Pembeli.cpp
#include "myclass/Pembeli.h"
#include <iostream>

std::string Pembeli::user()
{
    return "ini pembeli";
}

Pembeli::Pembeli(std::string nama, std::string gender, std::string noTelp, std::string alamat, int umur) : User(nama, gender, umur)
{
    this->noTelpPembeli = noTelp;
    this->alamatPembeli = alamat;
}

void Pembeli::infoPembeli()
{
    User::infoUser();
    std::cout << " tahun, Alamat : " << this->alamatPembeli << ", NoTelp : " << this->noTelpPembeli << std::endl;
}

void Pembeli::setNoTelpPembeli(std::string noTelp)
{
    this->noTelpPembeli = noTelp;
}

void Pembeli::setAlamatPembeli(std::string alamat)
{
    this->alamatPembeli = alamat;
}

std::string Pembeli::getNoTelpPembeli()
{
    return this->noTelpPembeli;
}

std::string Pembeli::getAlamatPembeli()
{
    return this->alamatPembeli;
}

File: Java/Java Kasus/src/com/myclass/Pembeli.java
package com.myclass;

public class Pembeli extends User {
    private String noTelp, alamat;

    public String displayUser() {
        return "Ini Pembeli";
    }

    public Pembeli(String nama, int umur, String jenisKelamin, String noTelp, String alamat) {
        super(nama, umur, jenisKelamin);
        this.noTelp = noTelp;
        this.alamat = alamat;
    }

    public void infoPembeli() {
        super.info();
        System.out.println(", No Telepon : " + this.noTelp + ", Alamat : " + this.alamat + ".\n");
    }

    public void setNoTelpPembeli(String noTelp) {
        this.noTelp = noTelp;
    }

    public void setAlamatPembeli(String alamat) {
        this.alamat = alamat;
    }

    public String getNoTelpPembeli() {
        return this.noTelp;
    }

    public String getAlamatPembeli() {
        return this.alamat;
    }

}
